srtcore/api.cpp

```diff
@@ -254,18 +254,24 @@ int CUDTUnited::startup()
 
    PacketFilter::globalInit();
 
-   //init CTimer::EventLock
-
    if (m_bGCStatus)
-      return true;
+      return 1;
 
    m_bClosing = false;
 
    setupMutex(m_GCStopLock, "GCStop");
    setupCond(m_GCStopCond, "GCStop");
    {
        ThreadName tn("SRT:GC");
-       pthread_create(&m_GCThread, NULL, garbageCollect, this);
+       try
+       {
+          SetThread(m_GCThread, garbageCollect, this);
```

Comment:
Again here returns the question as to whether it is reasonable to handle errors from this function by an exception. We define this function, we can decide how it does error handling. In our use cases it's actually more comfortable if it is handled by return value.